### PR TITLE
Windows: Update packaging tests

### PR DESCRIPTION
**Pull request by nicolecheetham**
Added windows compatibility to the test test_relocate_links
Added pytest not_on_windows markers to tests not intended to work on Windows and that needed a certain executable

### CODE REVIEW

**Review by johnwparent — DISMISSED** [review]

File: lib/spack/spack/test/packaging.py [hunk 1]
```diff
@@ -39,9 +40,8 @@
 )
 from spack.spec import Spec
 
-pytestmark = pytest.mark.not_on_windows("does not run on windows")
-
 
+@pytest.mark.not_on_windows("gpg not available on Windows")
```
Comment (scheibelp) [inline lib/spack/spack/test/packaging.py:46]: The `mock_gnupghome` test fixture should automatically skip this test if gpg is not available, were you seeing an issue without this? Once gpg is added with https://github.com/spack/spack/pull/41810, I think it would be useful to start running this test.

File: lib/spack/spack/test/packaging.py [hunk 2]
```diff
@@ -199,6 +200,7 @@ def test_needs_relocation():
     assert needs_binary_relocation("application", "x-mach-binary")
 
 
+@pytest.mark.not_on_windows("macos use not intended for Windows")
```
Comment (scheibelp) [inline lib/spack/spack/test/packaging.py]: I'm confused by this because it's also not intended for Linux: can you rephrase in terms of whatever part of this doesn't work on Windows? Could be something like "dylib manipulation logic in X does not run on Windows" (where X is whatever function in here fails on Windows).

**Review by scheibelp — CHANGES_REQUESTED** [review]

File: lib/spack/spack/test/packaging.py (hunk 1 above)
Comment (nicolecheetham) [reply lib/spack/spack/test/packaging.py:46]: With mock_gnupghome but without the not_on_windows pytest, the test results in an error message stating it can't bootstrap gpg.

File: lib/spack/spack/test/packaging.py (hunk 2 above)
Comment (nicolecheetham) [reply lib/spack/spack/test/packaging.py]: Okay. I just changed the pytest descriptions to be clearer.
